File: src/bot1_examples/nodes/simple_subscriber.py
#!/user/bin/env python3
import rospy
from std_msgs.msg import String

def msgcallback(msg):
    rospy.loginfo("The message recived %s", msg.data )


if __name__ == '__main__':
    rospy.init_node("simple_subscriber_py", anonymous=True)
    rospy.Subscriber("chatter", String, msgcallback)

    rospy.spin()

File: src/bot1_examples/nodes/simple_subscriber.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void msgcallback(const std_msgs::String::ConstPtr& msg)
{
    ROS_INFO("THE MESSAGE RECIVED: %s", msg->data.c_str());

}

int main(int argc, char** argv)
{
    ros::init(argc, argv,"simple_subscriber_cpp");
    ros::NodeHandle n;
    ros::Subscriber sub=n.subscribe("chatter", 10, msgcallback);
    ros::spin();




    return 0;
}